### DISCUSSION TITLE
Consumer pull last message for subject

### DISCUSSION THREAD

**Discussion by bisegni** [nats-io/nats.go#897] [Q&A] [answered]
Hi all, can NATS client to pull last message for a subject instead of pulling data from the initial offset specified?

**Comment by derekcollison** [comment, marked as answer]
If you are referring to JetStream and JetStream consumers yes.

**Reply by derekcollison** [reply]
DeliverPolicy: DeliverLastPerSubject

![image](https://user-images.githubusercontent.com/90097/152607207-3617d60f-61fe-43f6-a355-4d4875bde970.png)

**Reply by bisegni** [reply]
Hi thanks you for reply, I see that but I think this is for starting the read, but how can, at every read got only the last message?

**Reply by derekcollison** [reply]
You can specify MaxMsgsPer of 1 in the stream config.
